[2.0.x] Marlin does not compile to STM32 using default example config

### Compile Error
When I compile for STM32 with default example configuration the compilation fails:
```
Marlin\src\HAL\HAL_STM32F1\HAL_spi_Stm32f1.cpp:49:8: error: 'SPISettings' does not name a type
```
[Marlin2_stm32f1_buildlog.txt](https://github.com/MarlinFirmware/Marlin/files/1509898/Marlin2_stm32f1_buildlog.txt)

@victorpv latest changes doesnt compiles for me with PIO
Where StepperTimer and TempTimer were defined?
PS I have stm32f103RET board...

```
Compiling .pioenvs/STM32F1/src/src/HAL/HAL_LPC1768/include/digipot_mcp4451_I2C_routines.o
In file included from Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_Stm32f1.h:61:0,
from Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL.h:50,
from Marlin/src/HAL/HAL_LPC1768/../../inc/MarlinConfig.h:28,
from Marlin/src/HAL/HAL_LPC1768/LPC1768_Servo.cpp:63:
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h: In function 'void HAL_timer_set_count(uint8_t, hal_timer_t)':                                                                                                                                 
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:120:5: error: 'StepperTimer' was not declared in this scope                                                                                                                                    
StepperTimer.pause();
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:126:5: error: 'TempTimer' was not declared in this scope
TempTimer.pause();
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h: In function 'hal_timer_t HAL_timer_get_count(uint8_t)':
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:140:12: error: 'StepperTimer' was not declared in this scope                                                                                                                                   
temp = StepperTimer.getCompare(STEP_TIMER_CHAN);
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:143:12: error: 'TempTimer' was not declared in this scope
temp = TempTimer.getCompare(TEMP_TIMER_CHAN);
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h: In function 'void HAL_timer_set_current_count(uint8_t, hal_timer_t)':                                                                                                                         
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:154:26: error: 'StepperTimer' was not declared in this scope                                                                                                                                   
case STEP_TIMER_NUM: StepperTimer.setCount(count); break;
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:155:26: error: 'TempTimer' was not declared in this scope
case TEMP_TIMER_NUM: TempTimer.setCount(count); break;
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h: In function 'hal_timer_t HAL_timer_get_current_count(uint8_t)':                                                                                                                               
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:163:12: error: 'StepperTimer' was not declared in this scope                                                                                                                                   
temp = StepperTimer.getCount();
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:166:12: error: 'TempTimer' was not declared in this scope
temp = TempTimer.getCount();
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h: In function 'void HAL_timer_isr_prologue(uint8_t)':
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:180:5: error: 'StepperTimer' was not declared in this scope                                                                                                                                    
StepperTimer.pause();
^
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_timers_Stm32f1.h:186:5: error: 'TempTimer' was not declared in this scope
TempTimer.pause();
^
In file included from Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL.h:50:0,
from Marlin/src/HAL/HAL_LPC1768/../../inc/MarlinConfig.h:28,
from Marlin/src/HAL/HAL_LPC1768/LPC1768_Servo.cpp:63:
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_Stm32f1.h: At global scope:
Marlin/src/HAL/HAL_LPC1768/../../inc/../HAL/HAL_STM32F1/HAL_Stm32f1.h:157:12: warning: 'int freeMemory()' defined but not used [-Wunused-function]                                                                                                                          
static int freeMemory() {
^
Compiling .pioenvs/STM32F1/src/src/HAL/HAL_LPC1768/main.o
*** [.pioenvs/STM32F1/src/src/HAL/HAL_LPC1768/LPC1768_Servo.o] Error 1
```

--- a/Marlin/stepper.cpp
+++ b/Marlin/stepper.cpp
@@ -447,13 +447,10 @@ void Stepper::isr() {
   }
 
   // If there is no current block, attempt to pop one from the buffer
-  bool first_step = false;
   if (!current_block) {
     // Anything in the buffer?
     if ((current_block = planner.get_current_block())) {
       trapezoid_generator_reset();
-      TCNT1 = 0;  // make sure first pulse is not truncated
-      first_step = true;
 
       // Initialize Bresenham counters to 1/2 the ceiling
       counter_X = counter_Y = counter_Z = counter_E = -(current_block->step_event_count >> 1);
@@ -708,14 +705,8 @@ void Stepper::isr() {
   // Calculate new timer value
   if (step_events_completed <= (uint32_t)current_block->accelerate_until) {
 
-    if (first_step) {
-      acc_step_rate = current_block->initial_rate;
-      acceleration_time = 0;
-    }
-    else {
-      MultiU24X32toH16(acc_step_rate, acceleration_time, current_block->acceleration_rate);
-      acc_step_rate += current_block->initial_rate;
-    }
+    MultiU24X32toH16(acc_step_rate, acceleration_time, current_block->acceleration_rate);
+    acc_step_rate += current_block->initial_rate;
 
     // upper limit
     NOMORE(acc_step_rate, current_block->nominal_rate);

--- a/Marlin/stepper.h
+++ b/Marlin/stepper.h
@@ -359,6 +359,9 @@ class Stepper {
       OCR1A_nominal = calc_timer_interval(current_block->nominal_rate);
       // make a note of the number of step loops required at nominal speed
       step_loops_nominal = step_loops;
+      acc_step_rate = current_block->initial_rate;
+      acceleration_time = calc_timer_interval(acc_step_rate);
+      _NEXT_ISR(acceleration_time);
 
       #if ENABLED(LIN_ADVANCE)
         if (current_block->use_advance_lead) {

